cassandra-sink/src/main/java/org/springframework/cloud/stream/module/cassandra/sink/CassandraSinkConfiguration.java

```diff
@@ -0,0 +1,205 @@
+			cassandraMessageHandler.setWriteOptions(
+					new WriteOptions(cassandraSinkProperties.getConsistencyLevel(), cassandraSinkProperties.getRetryPolicy(), cassandraSinkProperties.getTtl()));
+		}
+		if (StringUtils.hasText(cassandraSinkProperties.getIngestQuery())) {
+			cassandraMessageHandler.setIngestQuery(cassandraSinkProperties.getIngestQuery());
+		}
+		else if (StringUtils.hasText(cassandraSinkProperties.getStatementExpression())) {
+			Expression expression = PARSER.parseExpression(cassandraSinkProperties.getStatementExpression());
+			cassandraMessageHandler.setStatementExpression(expression);
+		}
+		return cassandraMessageHandler;
+	}
+
+	private static boolean isUuid(String uuid) {
+		if (uuid.length() == 36) {
+			String[] parts = uuid.split("-");
+			if (parts.length == 5) {
+				if ((parts[0].length() == 8) && (parts[1].length() == 4) &&
+						(parts[2].length() == 4) && (parts[3].length() == 4) &&
+						(parts[4].length() == 12)) {
+					return true;
+				}
+			}
+		}
+		return false;
+	}
+
+
+	private static class PayloadToMatrixTransformer extends AbstractPayloadTransformer<Object, List<List<Object>>> {
+
+		private static final Pattern PATTERN = Pattern.compile(".+\\((.+)\\).+(?:values\\s*\\((.+)\\))");
+
+		private final ObjectMapper objectMapper = new ObjectMapper();
+
+		private final JsonObjectMapper<?, ?> jsonObjectMapper = new Jackson2JsonObjectMapper(this.objectMapper);
+
+		private final List<String> columns = new LinkedList<>();
+
+		private final ISO8601StdDateFormat dateFormat = new ISO8601StdDateFormat();
+
+		public PayloadToMatrixTransformer(String query) {
+			Matcher matcher = PATTERN.matcher(query);
+			if (matcher.matches()) {
+				String[] columns = StringUtils.delimitedListToStringArray(matcher.group(1), ",", " ");
+				String[] values = StringUtils.delimitedListToStringArray(matcher.group(2), ",", " ");
+				for (int i = 0; i < columns.length; i++) {
+					if (values[i].equals("?")) {
+						this.columns.add(columns[i]);
+					}
+				}
+			}
+			else {
+				throw new IllegalArgumentException("Invalid CQL insert query syntax: " + query);
+			}
+			this.objectMapper.configure(DeserializationFeature.ACCEPT_SINGLE_VALUE_AS_ARRAY, true);
+		}
+
+		@Override
+		@SuppressWarnings("unchecked")
+		protected List<List<Object>> transformPayload(Object payload) throws Exception {
+			if (payload instanceof List) {
+				return (List<List<Object>>) payload;
+			}
+			else {
+				List<Map<String, Object>> model = this.jsonObjectMapper.fromJson(payload, List.class);
+				List<List<Object>> data = new ArrayList<>(model.size());
+				for (Map<String, Object> entity : model) {
+					List<Object> row = new ArrayList<>(this.columns.size());
+					for (String column : this.columns) {
+						Object value = entity.get(column);
+						if (value instanceof String) {
+							String string = (String) value;
+							if (this.dateFormat.looksLikeISO8601(string)) {
+								synchronized (this.dateFormat) {
```

Comment:
Let me guess: because GC is more expensive than short `synchronized`.
Pay attention that its field is `final`, so it isn't counted by GC at all.